### DISCUSSION TITLE
Unable to Pull Basketball League Data for 2018

### DISCUSSION THREAD

**Discussion by jakeheffron** [cwendt94/espn-api#416] [ESPN API] [answered]
I'm unable to pull data specifically for 2018 for my private league. I get a "KeyError: '46'" which appears to stem from the _fetch_roster method. Error is below:

`---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_17272/4242162235.py in <module>
      6 for year in range(2017,2018):
----> 7     league = League(league_id=[my_league], year=year, swid="[my_swid]", espn_s2="[my_espn_s2]")

c:\users\jaket\appdata\local\programs\python\python39\lib\site-packages\espn_api\basketball\league.py in __init__(self, league_id, year, espn_s2, swid, fetch_league, debug)
     17 
     18         if fetch_league:
---> 19             self.fetch_league()
     20 
     21     def fetch_league(self):

c:\users\jaket\appdata\local\programs\python\python39\lib\site-packages\espn_api\basketball\league.py in fetch_league(self)
     21     def fetch_league(self):
     22         data = self._fetch_league()
---> 23         self._fetch_teams(data)
     24 
     25         self.BoxScoreClass = get_box_scoring_type_class(self.settings.scoring_type)

c:\users\jaket\appdata\local\programs\python\python39\lib\site-packages\espn_api\basketball\league.py in _fetch_teams(self, data)
     47         '''Fetch teams in league'''
     48         pro_schedule = self._get_all_pro_schedule()
---> 49         super()._fetch_teams(data, TeamClass=Team, pro_schedule=pro_schedule)
     50 
     51         # replace opponentIds in schedule with team instances

c:\users\jaket\appdata\local\programs\python\python39\lib\site-packages\espn_api\base_league.py in _fetch_teams(self, data, TeamClass, pro_schedule)
     62                     break
     63             roster = team_roster[team['id']]
---> 64             self.teams.append(TeamClass(team, roster=roster, member=member, schedule=schedule, year=seasonId, pro_schedule=pro_schedule))
     65 
     66         # sort by team ID

c:\users\jaket\appdata\local\programs\python\python39\lib\site-packages\espn_api\basketball\team.py in __init__(self, data, member, roster, schedule, year, **kwargs)
     30             self.logo_url = data['logo']
     31 
---> 32         self._fetch_roster(roster, year, kwargs.get('pro_schedule'))
     33         self._fetch_schedule(schedule)
     34 

c:\users\jaket\appdata\local\programs\python\python39\lib\site-packages\espn_api\basketball\team.py in _fetch_roster(self, data, year, pro_schedule)
     43 
     44         for player in roster:
---> 45             self.roster.append(Player(player, year, pro_schedule))
     46 
     47 

c:\users\jaket\appdata\local\programs\python\python39\lib\site-packages\espn_api\basketball\player.py in __init__(self, data, year, pro_team_schedule)
     41             if split['stats']:
     42                 if 'averageStats' in split.keys():
---> 43                     self.stats[id]['avg'] = {STATS_MAP[i]: split['averageStats'][i] for i in split['averageStats'].keys() if STATS_MAP[i] != ''}
     44                     self.stats[id]['total'] = {STATS_MAP[i]: split['stats'][i] for i in split['stats'].keys() if STATS_MAP[i] != ''}
     45                 else:

c:\users\jaket\appdata\local\programs\python\python39\lib\site-packages\espn_api\basketball\player.py in <dictcomp>(.0)
     41             if split['stats']:
     42                 if 'averageStats' in split.keys():
---> 43                     self.stats[id]['avg'] = {STATS_MAP[i]: split['averageStats'][i] for i in split['averageStats'].keys() if STATS_MAP[i] != ''}
     44                     self.stats[id]['total'] = {STATS_MAP[i]: split['stats'][i] for i in split['stats'].keys() if STATS_MAP[i] != ''}
     45                 else:

KeyError: '46'`

**Comment by cwendt94** [comment, marked as answer]
Thanks for reporting this, I have a PR up now to handle unknown stat ids!

**Reply by cwendt94** [reply]
should be fixed in v0.28.0!
